nodejs/week1/mealSharing/src/backend/routes/meals.js

```diff
@@ -0,0 +1,32 @@
+
+
+const meal = require("../data/meals.json");
+const review = require("../data/reviews.json");
+
+let mealWithReview = [],
+mealWithoutReview = [];
+
+
+const allMeals = () =>{
+for (let i = 0; i < meal.length; i++) {
+  for (let r = 0 ; r < review.length ; r++ ){
+    if(meal[i].id == review[r].mealId){
+      mealWithReview.push(review[r]);
+      // @ts-ignore
+      meal[i].reviews = mealWithReview;
+      
+    }
+    else{
+      // @ts-ignore
+      meal[i].reviews = mealWithoutReview;
+
+    }
+  }
+  
+  
+}
+}
```

Comment:
You have a bug here ❕  Try adding another review in reviews.json, for example with `"mealId": 7` and then check which meals have reviews at `/meals`. 

Instead of having nested `for` loops, you can replace the inner  loop with `reviews.filter(...)` to get all reviews that have `mealId` equal to `meal[i].id`.